diff --git a/node_modules/@digitalcredentials/vc/lib/CredentialIssuancePurpose.js b/node_modules/@digitalcredentials/vc/lib/CredentialIssuancePurpose.js
index c7fb324..1e70b76 100644
--- a/node_modules/@digitalcredentials/vc/lib/CredentialIssuancePurpose.js
+++ b/node_modules/@digitalcredentials/vc/lib/CredentialIssuancePurpose.js
@@ -67,7 +67,10 @@ class CredentialIssuancePurpose extends AssertionProofPurpose {
 
       const issuerId = typeof issuer[0] === 'string' ? issuer[0] : issuer[0].id;
 
-      if(result.controller.id !== issuerId) {
+      const { document: issuerDoc } = await documentLoader(issuerId)
+      const reroutedIsuserId = issuerDoc.id
+
+      if(result.controller.id !== reroutedIsuserId) {
         throw new Error(
           'Credential issuer must match the verification method controller.');
       }
